### ISSUE TITLE
No error reported when upload to S3 fails

### ISSUE THREAD

**Issue by buxm** [tiredofit/docker-db-backup#199]
### Summary

When the upload to S3 fails, no error is reported in the logs and `$1` parameter for post backup scripts is `0` (i.e. success).

### Steps to reproduce

1. Create Pod template/docker compose file for a manual run (`MODE=MANUAL`, `MANUAL_RUN_FOREVER=FALSE`) backup of a database with: `BACKUP_LOCATION=S3`; incorrect parameters to access the S3 destination (so that the upload to S3 fails); a post script that checks the `$1` paramenter
2. Run the pod/container
3. Check the logs of the pod/container
4. The logs of the pod container say:
> `2023-01-11.21:18:02 [INFO] ** [db-backup] Backup routines finish time: 2023-01-11 21:18:02 GMT with overall exit code 0`
5. The `$1` parameter of the post script contains `0`

### What is the expected *correct* behavior?

1. The logs of the pod/container report an error when uploading to S3
2. In the post script the `$1` contains a value different from `0`

### Relevant logs and/or screenshots

```
[...]
2023-01-11.21:36:10 [INFO] /etc/services.available/10-db-backup/run ** [db-backup] DB Backup of 'mysql_database_mariadb_20230111-213610.sql.xz' completed successfully
/assets/functions/00-container: line 850: /var/log/container//container.log: No such file or directory
+ generate_checksum
+ var_true TRUE
+ '[' true = true ']'
+ '[' 0 = 0 ']'
+ case "${CHECKSUM,,}" in
+ checksum_command=md5sum
+ checksum_extension=md5
+ print_notice 'Generating MD5 for '\''mysql_database_mariadb_20230111-213610.sql.xz'\'''
+ output_off
+ '[' true = true ']'
+ set +x
2023-01-11.21:36:10 [NOTICE] /etc/services.available/10-db-backup/run ** [db-backup] Generating MD5 for 'mysql_database_mariadb_20230111-213610.sql.xz'
/assets/functions/00-container: line 876: /var/log/container//container.log: No such file or directory
+ cd /tmp/backups
+ md5sum mysql_database_mariadb_20230111-213610.sql.xz
++ awk ' { print $1}'
++ md5sum mysql_database_mariadb_20230111-213610.sql.xz
+ checksum_value=02e4a2063f7c3cc30669c925d18687ac
+ print_debug 'MD5: 02e4a2063f7c3cc30669c925d18687ac - mysql_database_mariadb_20230111-213610.sql.xz'
+ output_off
+ '[' true = true ']'
+ set +x
2023-01-11.21:36:10 [DEBUG] /etc/services.available/10-db-backup/run ** [db-backup] MD5: 02e4a2063f7c3cc30669c925d18687ac - mysql_database_mariadb_20230111-213610.sql.xz
/assets/functions/00-container: line 800: /var/log/container//container.log: No such file or directory
+ move_dbbackup
+ '[' 0 = 0 ']'
++ stat -c%s /tmp/backups/mysql_database_mariadb_20230111-213610.sql.xz
+ dbbackup_size=4696
++ date -r /tmp/backups/mysql_database_mariadb_20230111-213610.sql.xz +%s
+ dbbackup_date=1673472970
+ case "${SIZE_VALUE,,}" in
+ SIZE_VALUE=-h
+ '[' -h = 1 ']'
++ awk '{ print $1}'
++ du -h /tmp/backups/mysql_database_mariadb_20230111-213610.sql.xz
+ filesize=8.0K
+ print_notice 'Backup of mysql_database_mariadb_20230111-213610.sql.xz created with the size of 8.0K'
+ output_off
+ '[' true = true ']'
+ set +x
2023-01-11.21:36:10 [NOTICE] /etc/services.available/10-db-backup/run ** [db-backup] Backup of mysql_database_mariadb_20230111-213610.sql.xz created with the size of 8.0K
/assets/functions/00-container: line 876: /var/log/container//container.log: No such file or directory
+ case "${BACKUP_LOCATION,,}" in
+ print_debug 'Moving backup to S3 Bucket'
+ output_off
+ '[' true = true ']'
+ set +x
2023-01-11.21:36:10 [DEBUG] /etc/services.available/10-db-backup/run ** [db-backup] Moving backup to S3 Bucket
/assets/functions/00-container: line 800: /var/log/container//container.log: No such file or directory
+ '[' -n XXXXXXXXXXXXXXXXXXX ']'
+ '[' -n XXXXXXXXXXXXXXXXXXX ']'
+ export AWS_ACCESS_KEY_ID=XXXXXXXXXXXXXXXXXXX
+ AWS_ACCESS_KEY_ID=XXXXXXXXXXXXXXXXXXX
+ export AWS_SECRET_ACCESS_KEY=XXXXXXXXXXXXXXXXXXX
+ AWS_SECRET_ACCESS_KEY=XXXXXXXXXXXXXXXXXXX
+ export AWS_DEFAULT_REGION=eu-west-1
+ AWS_DEFAULT_REGION=eu-west-1
+ '[' -f '' ']'
+ var_true FALSE
+ '[' false = true ']'
+ '[' false = yes ']'
+ [[ -n '' ]]
+ silent aws s3 cp /tmp/backups/mysql_database_mariadb_20230111-213610.sql.xz s3://my-bucket/my-prefix/mysql_database_mariadb_20230111-213610.sql.xz --sse aws:kms
+ '[' TRUE = true ']'
+ '[' '' = true ']'
+ '[' debug = debug ']'
+ aws s3 cp /tmp/backups/mysql_database_mariadb_20230111-213610.sql.xz s3://my-bucket/my-prefix/mysql_database_mariadb_20230111-213610.sql.xz --sse aws:kms
upload failed: ./mysql_database_mariadb_20230111-213610.sql.xz to s3://my-bucket/my-prefix/mysql_database_mariadb_20230111-213610.sql.xz An error occurred (SignatureDoesNotMatch) when
calling the PutObject operation: The request signature we calculated does not match the signature you provided. Check your key and signing method.
+ var_true TRUE
+ '[' true = true ']'
+ silent aws s3 cp /tmp/backups/mysql_database_mariadb_20230111-213610.sql.xz.md5 s3://my-bucket/my-prefix/ --sse aws:kms
+ '[' TRUE = true ']'
+ '[' '' = true ']'
+ '[' debug = debug ']'
+ aws s3 cp /tmp/backups/mysql_database_mariadb_20230111-213610.sql.xz.md5 s3://my-bucket/my-prefix/ --sse aws:kms
upload failed: ./mysql_database_mariadb_20230111-213610.sql.xz.md5 to s3://my-bucket/my-prefix/mysql_database_mariadb_20230111-213610.sql.xz.md5 An error occurred (SignatureDoesNotMatc
h) when calling the PutObject operation: The request signature we calculated does not match the signature you provided. Check your key and signing method.
+ rm -rf /tmp/backups/mysql_database_mariadb_20230111-213610.sql.xz.md5
+ rm -rf /tmp/backups/mysql_database_mariadb_20230111-213610.sql.xz
+ rm -rf '/tmp/backups/*'
+ post_dbbackup database
++ date +%s
+ dbbackup_finish_time=1673472971
++ echo 1
+ dbbackup_total_time=1
+ var_true FALSE
+ '[' false = true ']'
+ '[' false = yes ']'
+ '[' -n '' ']'
+ '[' -d /assets/custom-scripts/ ']'
+ '[' -d /assets/scripts/post/ ']'
+ dir_notempty /assets/scripts/post/
++ ls -A /assets/scripts/post/
+ '[' -n '..2023_01_11_21_36_08.3164422953
..data
db-backup-post.sh' ']'
++ find /assets/scripts/post/ -name '*.sh' -type f
+ for f in $(find ${SCRIPT_LOCATION_POST} -name \*.sh -type f)
+ var_true ''
+ '[' '' = true ']'
+ '[' '' = yes ']'
+ '[' -x /assets/scripts/post/..2023_01_11_21_36_08.3164422953/db-backup-post.sh ']'
+ print_notice 'Executing post backup custom script : '\''/assets/scripts/post/..2023_01_11_21_36_08.3164422953/db-backup-post.sh'\'''
+ output_off
+ '[' true = true ']'
+ set +x
2023-01-11.21:36:11 [NOTICE] /etc/services.available/10-db-backup/run ** [db-backup] Executing post backup custom script : '/assets/scripts/post/..2023_01_11_21_36_08.3164422953/db-backup-post.s
h'
/assets/functions/00-container: line 876: /var/log/container//container.log: No such file or directory
+ /assets/scripts/post/..2023_01_11_21_36_08.3164422953/db-backup-post.sh 0 mysql mariadb database 1673472970 1673472971 1 mysql_database_mariadb_20230111-213610.sql.xz 8.0K 02e4a2063f7c3cc30669
c925d18687ac
mysql Backup Completed on mariadb for database on 1673472970 ending 1673472971 for a duration of 1 seconds. Filename: mysql_database_mariadb_20230111-213610.sql.xz Size: 8.0K bytes
++ awk '{printf "Hours: %d Minutes: %02d Seconds: %02d", $1/3600, ($1/60)%60, $1%60}'
++ echo 1
+ print_notice 'DB Backup for '\''database'\'' time taken: Hours: 0 Minutes: 00 Seconds: 01'
+ output_off
+ '[' true = true ']'
+ set +x
2023-01-11.21:36:11 [NOTICE] /etc/services.available/10-db-backup/run ** [db-backup] DB Backup for 'database' time taken: Hours: 0 Minutes: 00 Seconds: 01
/assets/functions/00-container: line 876: /var/log/container//container.log: No such file or directory
+ unset s3_ssl
+ unset s3_ca_cert
++ date +%s
+ backup_finish_time=1673472971
++ echo 1
+ backup_total_time=1
+ '[' -z '' ']'
+ master_exit_code=0
++ date -d @1673472971 '+%Y-%m-%d %T %Z'
+ print_info 'Backup routines finish time: 2023-01-11 21:36:11 GMT with overall exit code 0'
+ output_off
+ '[' true = true ']'
+ set +x
2023-01-11.21:36:11 [INFO] /etc/services.available/10-db-backup/run ** [db-backup] Backup routines finish time: 2023-01-11 21:36:11 GMT with overall exit code 0
/assets/functions/00-container: line 850: /var/log/container//container.log: No such file or directory
++ awk '{printf "Hours: %d Minutes: %02d Seconds: %02d", $1/3600, ($1/60)%60, $1%60}'
++ echo 1
+ print_notice 'Backup routines time taken: Hours: 0 Minutes: 00 Seconds: 01'
+ output_off
+ '[' true = true ']'
+ set +x
2023-01-11.21:36:11 [NOTICE] /etc/services.available/10-db-backup/run ** [db-backup] Backup routines time taken: Hours: 0 Minutes: 00 Seconds: 01
/assets/functions/00-container: line 876: /var/log/container//container.log: No such file or directory
+ cleanup_old_data
+ '[' -n '' ']'
+ var_true TRUE
+ '[' true = true ']'
+ print_debug 'Exiting due to manual mode'
+ output_off
+ '[' true = true ']'
+ set +x
2023-01-11.21:36:11 [DEBUG] /etc/services.available/10-db-backup/run ** [db-backup] Exiting due to manual mode
/assets/functions/00-container: line 800: /var/log/container//container.log: No such file or directory
+ exit 0
```
### Environment

- Image version / tag:  3.7.3
- Host OS: Oracle Kubernetes Cluster/Oracle-Linux-8.6-aarch64-2022.12.06-0

<details>
<summary>backup-job-template.yml</summary>

```
apiVersion: v1
kind: Secret
metadata:
  name: database-backup-storage-secondary
type: Opaque
stringData:
  S3_CERT_SKIP_VERIFY: "FALSE"
  S3_BUCKET: my-bucket
  S3_PATH: my-prefix
  S3_REGION: eu-west-1
  S3_EXTRA_OPTS: --sse aws:kms
  S3_KEY_ID: XXXXXXXXXXXXXXX
  S3_KEY_SECRET: XXXXXXXXXXXXXXXXXXXXXX
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: database-backup-post-scripts-secondary
  labels:
    app: database
    tier: db-backup
data:
  db-backup-post.sh: |
    #!/bin/bash
     #### Example Post Script
     #### $1=EXIT_CODE (After running backup routine)
     #### $2=DB_TYPE (Type of Backup)
     #### $3=DB_HOST (Backup Host)
     #### #4=DB_NAME (Name of Database backed up
     #### $5=BACKUP START TIME (Seconds since Epoch)
     #### $6=BACKUP FINISH TIME (Seconds since Epoch)
     #### $7=BACKUP TOTAL TIME (Seconds between Start and Finish)
     #### $8=BACKUP FILENAME (Filename)
     #### $9=BACKUP FILESIZE
     #### $10=HASH (If CHECKSUM enabled)
    
    if [ $1 ]; then
      echo ${2} Backup Completed on ${3} for ${4} on ${5} ending ${6} for a duration of ${7} seconds. Filename: ${8} Size: ${9} bytes
    else
      echo BACKUP ERROR!!!
    fi
    
---
apiVersion: batch/v1
kind: Job
metadata:
  name: database-backup-job-secondary
spec:
  backoffLimit: 3
  ttlSecondsAfterFinished: 600
  template:
    spec:
      containers:
      - name: database-backup
        image: tiredofit/db-backup:latest
        
        command:
        - "/etc/services.available/10-db-backup/run"
        
        volumeMounts:
          - mountPath: /tmp/backups
            name: database-backup-tmp-volume
          - mountPath: /assets/scripts/post
            name: database-db-backup-post-scripts
         
        env:
        - name: TIMEZONE
          value: Europe/Rome
        - name: BACKUP_LOCATION
          value: S3
        - name: MODE
          value: MANUAL
        - name: MANUAL_RUN_FOREVER
          value: "FALSE"
        - name: SIZE_VALUE
          value: megabytes
          
        - name: COMPRESSION
          value: XZ
        - name: COMPRESSION_LEVEL
          value: "9"
        
        - name: CONTAINER_ENABLE_SCHEDULING
          value: "FALSE"
        - name: CONTAINER_ENABLE_MONITORING
          value: "FALSE"
        
        - name: DEBUG_MODE
          value: "TRUE"
          
        - name: DB_TYPE
          value: mysql
        - name: DB_HOST
          value: mariadb
        - name: DB_NAME
          valueFrom:
            secretKeyRef:
              name: database-db-secrets
              key: MYSQL_DATABASE
        - name: DB_USER
          valueFrom:
            secretKeyRef:
              name: database-db-secrets
              key: MYSQL_USER
        - name: DB_PASS
          valueFrom:
            secretKeyRef:
              name: database-db-secrets
              key: MYSQL_PASSWORD
        envFrom:
        - secretRef:
            name: database-backup-storage-secondary

      volumes:
      - name: database-backup-tmp-volume
        emptyDir: {}
      - name: database-db-backup-post-scripts
        configMap:
          name: database-backup-post-scripts-secondary
          defaultMode: 0500
      restartPolicy: Never
```
</details>

**Comment by alwynpan** [comment]
Create PR and tested working.

<img width="1197" alt="image" src="https://github.com/tiredofit/docker-db-backup/assets/9387800/f65bc1b2-198b-409f-9381-a1c152123afa">

Post backup script can correctly capture the error and act accordingly.

<img width="1197" alt="image" src="https://github.com/tiredofit/docker-db-backup/assets/9387800/6bdcbbc1-646f-44ab-9fac-ef235cb6d7d0">

**Comment by buxm** [comment]
I've tested the fix using release `3.9.6` and confirm it works as expected.
Many thanks!!! :-)
